helix-term/src/ui/statusline.rs

```diff
@@ -330,6 +330,12 @@ where
     F: Fn(&mut RenderContext, String, Option<Style>) + Copy,
 {
     let file_type = context.doc.language_id().unwrap_or("text");
```

Comment:
Somewhat unrelated to this change, I think this might be better off as the `scope` field from `languages.toml` which is `context.doc.language().unwrap_or("text")`. That way Rust is "source.rust", toml is "source.toml", html is "text.html.basic", etc.

Reply:
Do you suggest any changes to this PR? My opinion is that we should have a function whose sole purpose is to return the language. Right now there are several possible ways to do it, each with their own issues.

Reply:
> we should have a function whose sole purpose is to return the language

The trouble is: what is the language? We could say it's the `scope` or we could say it's the `name` (language_id but not `language_id()`, see also https://github.com/helix-editor/helix/pull/3338). It depends on what we want the public API of language names to be: is the `LanguageConfiguration.language_id` an implementation detail or is it ok for configurations to use them?

I'd be hesitant to merge this PR until we figure that out since this PR introduces a way for configurations to depend on language names/ids/scopes.

@archseer what do you think?

Reply:
This PR merely provides a mapping from the current language reporting functionality so that we can use custom strings that don't take up 1/4 of the status line. I'd be really disappointed if this PR was held up as a result of a feature that is still unimplemented.